h/api/auth.py

```diff
@@ -13,3 +18,13 @@ def get_user(request):
                 consumer = Consumer(key)
                 return User(userid, consumer, False)
     return None
+
+
+def validate_permissions(request, permissions):
+    """Validate that the annotation's permissions are within the principals."""
```

Comment:
This needs a better docstring. Maybe something more specific, like `"""Raise HTTPUnauthorized if any of the principals in permissions['read'] are not in effective_principals."""` And maybe a further description part that describes the purpose of this, that it prevents any user from making an annotation in a group that they're not a member of.